astropy/cosmology/parameter/_converter.py

```diff
@@ -3,23 +3,43 @@
 from __future__ import annotations
 
 from collections.abc import Callable
-from typing import Any
+from typing import TYPE_CHECKING, Any, TypeAlias, TypeVar, overload
```

Comment:
We should turn the ruff check on.

Reply:
I had a quick look at this. Just turning on the setting caused problems in a few places because of [dataclasses.KW_ONLY](https://docs.python.org/3/library/dataclasses.html#dataclasses.KW_ONLY).

Reply:
That is unfortunate! Ruff should be made aware of this. `KW_ONLY` can be special-cased.

Reply:
You are much more familiar with `dataclasses`, maybe you should try turning on the Ruff setting. Maybe the problems can be solved.

Reply:
I opened #16647, so am resolving the conversation in this PR.